airflow/providers/docker/decorators/docker.py

```diff
@@ -0,0 +1,459 @@
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import base64
+import inspect
+import os
+import pickle
+from tempfile import TemporaryDirectory
+from textwrap import dedent
+from typing import Callable, Dict, Iterable, List, Optional, TypeVar, Union
+
+from airflow.decorators.base import DecoratedOperator, task_decorator_factory
+from airflow.providers.docker.operators.docker import DockerOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.utils.python_virtualenv import remove_task_decorator, write_python_script
+
+
+def _generate_decode_command(env_var, file):
+    return (
+        f'python -c "import os; import base64;'
+        f' x = base64.b64decode(os.environ[\\"{env_var}\\"]);'
+        f' f = open(\\"{file}\\", \\"wb\\"); f.write(x);'
+        f' f.close()"'
+    )
+
+
+def _b64_encode_file(filename):
+    with open(filename, "rb") as file_to_encode:
+        return base64.b64encode(file_to_encode.read())
+
+
+class _DockerDecoratedOperator(DecoratedOperator, DockerOperator):
+    """
+    Wraps a Python callable and captures args/kwargs when called for execution.
+
+    :param python_callable: A reference to an object that is callable
+    :type python_callable: python callable
+    :param op_kwargs: a dictionary of keyword arguments that will get unpacked
+        in your function (templated)
+    :type op_kwargs: dict
+    :param op_args: a list of positional arguments that will get unpacked when
+        calling your callable (templated)
+    :type op_args: list
+    :param multiple_outputs: if set, function return value will be
+        unrolled to multiple XCom values. Dict will unroll to xcom values with keys as keys.
+        Defaults to False.
+    :type multiple_outputs: bool
+    """
+
+    template_fields = ('op_args', 'op_kwargs')
+    template_fields_renderers = {"op_args": "py", "op_kwargs": "py"}
+
+    # since we won't mutate the arguments, we should just do the shallow copy
+    # there are some cases we can't deepcopy the objects (e.g protobuf).
+    shallow_copy_attrs = ('python_callable',)
+
+    @apply_defaults
+    def __init__(
+        self,
+        **kwargs,
+    ) -> None:
+        self.string_args = [1, 2, 1]
+        self._output_filename = ""
+        command = "dummy command"
+        self.pickling_library = pickle
```

Comment:
Should support dill/anything else the same way VirtualEnv operator does please.

Reply:
I'm going to do pickle and dill for now as those are the two options that our virtualenv operator supports. Not sure about what others there are.